pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -950,19 +965,31 @@ func TestGetVerifiedTaskFunc_VerifyError(t *testing.T) {
 			}
 			fn := resources.GetVerifiedTaskFunc(ctx, k8sclient, tektonclient, tc.requester, tr, tr.Spec.TaskRef, "", "default", "default", vps)
 
-			resolvedTask, source, err := fn(ctx, taskRef.Name)
+			resolvedTask, clusterTask, source, err := fn(ctx, taskRef.Name)
 
 			if !errors.Is(err, tc.expectedErr) {
 				t.Errorf("GetVerifiedTaskFunc got %v but want %v", err, tc.expectedErr)
 			}
 
-			if d := cmp.Diff(tc.expected, resolvedTask); d != "" {
-				t.Errorf("resolvedTask did not match: %s", diff.PrintWantGot(d))
-			}
-
 			if source != nil {
 				t.Errorf("source is: %v but want is nil", source)
 			}
+
+			if clusterTask != nil {
+				if d := cmp.Diff(tc.expected, clusterTask); d != "" {
+					t.Errorf("resolved ClusterTask did not match: %s", diff.PrintWantGot(d))
+				}
+				return
+			}
+
+			// TODO
+			if resolvedTask == tc.expected {
+				return
+			}
+
+			if d := cmp.Diff(resolvedTask, tc.expected); d != "" {
```

Comment:
Could you please help me to understand why the nil pointers here are not asserted equal with nil tc.expected here even with the `cmpopts.EquateEmpty`?
cc @Yongxuanzhang

Reply:
the previous test cases are all comparing with nil, so I think it should work?

Reply:
Yep I also thought the same. but was stuck here for a bit :/

https://prow.tekton.dev/view/gs/tekton-prow/pr-logs/pull/tektoncd_pipeline/6452/pull-tekton-pipeline-unit-tests/1641051864147431424

Reply:
need to change the expected from `runtime.Object` to `*v1beta1.Task`

Reply:
Could we check if it makes sense to cast the expected nil as (*v1beta1.Task) while keeping the expected as `runtime.Object` for being more generic in the test case struct?

Reply:
We changed VerifyTask's in this PR to pass task not taskobject, so in tests we should test task

Reply:
Thanks for the pointer! updated.